### DISCUSSION TITLE
No obvious way to clear the plot

### DISCUSSION THREAD

**Discussion by yurivict** [alandefreitas/matplotplusplus#225] [Ideas]
I need to show multiple plots in the same process.

Adding plots and calling show() again shows plot with all previous drawings combined.

There is no "erase()" or "clear()" or "reset()" function and every testcase in the docs has a main function that exits immediately after the plot.

Could you please add a paragraph to the documentation on how to show consecutive plots in the sane process?

**Comment by alandefreitas** [comment]
You either need to create [multiplots](https://alandefreitas.github.io/matplotplusplus/appearance/multiplot/hold/) or work with multiple [figure](https://alandefreitas.github.io/matplotplusplus/appearance/figure-object/) objects.

`show()` and `draw()` and meant to just show what's in the figure for member functions, or the current figure for the free function. They have nothing to do with clearing the plot.

The usual way to clear the plot in these libraries is to [leave quiet mode](https://github.com/alandefreitas/matplotplusplus/blob/3fd7593f3bf98ca60fdf3db17073ea8bbe161263/source/matplot/core/figure_type.h#L167), and then plot something else. 

If you want to edit the objects directly, the axes have a member function to [clear](https://github.com/alandefreitas/matplotplusplus/blob/3fd7593f3bf98ca60fdf3db17073ea8bbe161263/source/matplot/core/axes_type.h#L594) their object and a [children](https://github.com/alandefreitas/matplotplusplus/blob/3fd7593f3bf98ca60fdf3db17073ea8bbe161263/source/matplot/core/axes_type.h#L589) function to get its children so you can edit their objects. You can also access the figure [children](https://github.com/alandefreitas/matplotplusplus/blob/3fd7593f3bf98ca60fdf3db17073ea8bbe161263/source/matplot/core/figure_type.h#L129) and edit, or clear, them. There's no clear function to do that in one step. This could be useful, to maintain symmetry with the axes. It just hasn't been requested. A small PR to improve this is welcome.

About the docs, they need improvements everywhere. There's no single issue for that, but any PR improving the docs here and there are also welcome.

**Comment by yurivict** [comment]
```gca()->children({ });``` clears the plot, but this looks like a hack.

--------------

> A small PR to improve this is welcome.

Sorry, I don't have resources for a PR at this time.
